components/canvas/hooks/useCanvasType.tsx

```diff
@@ -0,0 +1,18 @@
+import { createContext, PropsWithChildren, useContext } from "react";
+
+export type CanvasType = "main" | "select_process_card";
+
+const CanvasTypeContext = createContext<CanvasType>("main");
+
+export const useCanvasType = () => useContext(CanvasTypeContext);
```

Comment:
What are your thoughts on returning the canvas-related variables in this hook. For example instead of having useCanAddPQIR we would have 
export const useCanvasType = () => {
  const canvasType = useContext(CanvasTypeContext);
  const canAddPQIR = canvasType === "main";
  return { canAddPQIR }
}
And also adding the code from the other custom hooks in here instead of having multiple files?

Reply:
That is another approach, but the current approach's idea is to add separation of concern into the code.

`useCanvasType` will simply let you figure out which canvas type context you are in.
`useCanAddPQIR` will simply let you know if you should be able to add PQIR in the given context.

One could imagine that in the future, `useCanAddPQIR` could be affected by more variables than just `canvasType`.

By coupling them together into the same hook, we risk creating a hook that eventually ends up being a bloated mess as we add more dependencies into the logic.